BuildingsCreator/BuildingsCreator/HashTable/HashTable.cs

```diff
@@ -0,0 +1,72 @@
+﻿
+using System.Collections.Generic;
+
+namespace BuildingsCreator.HashTable
+{
+    /// <summary> Класс описывающий Хэш-таблицу. </summary>
+    /// <typeparam name="T"> Класс содержащийся в таблице. </typeparam>
+    class HashTable<T> where T : class
+    {
+        //т.к. номера зданий уникальны, то достаточно простого массива, а не List<T>[]
+        private static T[] items;
+
+        /// <summary> Создать экземпляр класса HashTable. </summary>
+        /// <param name="size"> Размер таблицы. </param>
+        public HashTable(int size)
+        {
+            items = new T[size];
+        }
+
+        #region HashTableMethodth
+
+        /// <summary> Добавить экземпляр класса в таблицу. </summary>
+        /// <param name="item"> Класс T. </param>
+        /// <param name="key"> Ключ для записи. </param>
+        public static void Add(T item, int key)
+        {
+            var hashKey = GetHash(key);
+            items[hashKey] = item;
+        }
+
+        /// <summary> Получить экзмпляр класса Т по ключу. </summary>
+        /// <param name="key"> Ключ для поиска значения. </param>
+        /// <returns> Экземпляр класса Т. </returns>
+        public static T Get(int key)
+        {
+            var hashKey = GetHash(key);
+            return items[hashKey];
+        }
+
+        /// <summary> Проверить наличие в таблице по ключу. </summary>
+        /// <param name="key"> Ключ для поиска экземпляра. </param>
+        /// <returns> Наличие в таблице. </returns>
+        public static bool Search(int key)
+        {
+            var hashKey = GetHash(key);
+            var item = items[hashKey];
+
+            if (item == null)
+                return false;
+            else
+                return true;
```

Comment:
Индусский код detected!
```suggestion
            return item != null;
```